plugin/nns/nns.go

```diff
@@ -0,0 +1,82 @@
+package nns
+
+import (
+	"context"
+	"fmt"
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+	nns "github.com/nspcc-dev/neo-go/examples/nft-nd-nns"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/client"
+	"strings"
+)
+
+type NNS struct {
+	Next   plugin.Handler
+	Client *client.Client
+	CS     *state.Contract
+	Log    clog.P
+}
+
+// ServeDNS implements the plugin.Handler interface. This method gets called when example is used
+// in a Server.
+func (n NNS) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+	if state.QType() != dns.TypeTXT {
+		return plugin.NextOrFailure(n.Name(), n.Next, ctx, w, r)
+	}
+
+	s, err := n.resolve(state.QName())
+	if err != nil {
+		n.Log.Warning(err)
+		return plugin.NextOrFailure(n.Name(), n.Next, ctx, w, r)
+	}
+
+	m := new(dns.Msg)
+	m.SetReply(r)
+	hdr := dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeTXT, Class: state.QClass(), Ttl: 0}
+	m.Answer = []dns.RR{&dns.TXT{Hdr: hdr, Txt: []string{s}}}
+
+	w.WriteMsg(m)
+	return dns.RcodeSuccess, nil
+}
+
+// Name implements the Handler interface.
+func (n NNS) Name() string { return pluginName }
+
+func (n NNS) resolve(name string) (string, error) {
+	name = strings.TrimSuffix(name, ".")
+	return n.Client.NNSResolve(n.CS.Hash, name, nns.TXT)
+}
+
+func (n NNS) Transfer(zone string, serial uint32) (<-chan []dns.RR, error) {
+	s, err := n.resolve(zone)
+	if err != nil {
+		n.Log.Warning(err)
+		return nil, err
+	}
+
+	soa, err := dns.NewRR(fmt.Sprintf("%s IN SOA nspcc.ru info.nspcc.ru %d 7200 3600 1209600 3600", zone, serial))
```

Comment:
I suppose transfer should lookup zone in the NNS contract and do the transfer based on the SOA record from it. 

I guess NNS contract is not ready for that yet. Therefore I propose:
- define list of things that are missing in NNS contract for zone transfer,
- implement it in nns contract in neofs-contract repository,
- support it there.

/cc @realloc